packages/strapi-plugin-graphql/services/schema-generator.js

```diff
@@ -67,56 +69,63 @@ const generateSchema = () => {
       ${definition}
       ${shadowCRUD.definition}
       ${polymorphicSchema.definition}
-
       ${Types.addInput()}
       
       ${PublicationState.definition}
-
       type AdminUser {
         id: ID!
         username: String
         firstname: String!
         lastname: String!
       }
-
       type Query {
         ${queryFields}
         ${query}
       }
-
       type Mutation {
         ${mutationFields}
         ${mutation}
       }
-
       ${scalarDef}
     `;
 
   // Build schema.
-  if (strapi.config.environment !== 'production') {
-    // Write schema.
-    const schema = makeExecutableSchema({
-      typeDefs,
-      resolvers,
-    });
+  const schema = makeExecutableSchema({
+    typeDefs,
+    resolvers,
+  });
 
-    writeGenerateSchema(graphql.printSchema(schema));
+  const generatedSchema = filterDisabledResolvers(schema, extraResolvers);
+
+  if (strapi.config.environment !== 'production') {
+    writeGenerateSchema(generatedSchema);
   }
 
-  return {
-    typeDefs: gql(typeDefs),
-    resolvers,
-  };
+  return isFederated ? getFederatedSchema(generatedSchema, resolvers) : generatedSchema;
 };
 
+const getFederatedSchema = (schema, resolvers) =>
+  buildFederatedSchema([{ typeDefs: gql(printSchema(schema)), resolvers }]);
```

Comment:
I think you might be able to only use printSchema

Reply:
this is not possible